src/main/java/dev/breeze/settlements/test/TestCommandHandler.java

```diff
@@ -72,4 +75,48 @@ public boolean onCommand(@Nonnull CommandSender sender, @Nonnull Command command
         return true;
     }
 
+    @Override
+    public @Nullable List<String> onTabComplete(@NotNull CommandSender sender, @NotNull Command command, @NotNull String label, @NotNull String[] args) {
+        List<String> tabComplete = new ArrayList<>();
+        if (!(sender instanceof Player)) {
+            return tabComplete;
+        }
+
+        if (args.length == 1) {
+            tabComplete.add("BaseVillager");
+            tabComplete.add("Armorstand");
+            tabComplete.add("Advacement");
+            tabComplete.add("ToggleDebug");
+        }
+        return tabComplete.stream().filter(completion -> completion.toLowerCase().contains(args[args.length - 1].toLowerCase())).toList();
+    }
+
+    public static void baseVillager(Block target, String[] args) {
+        new BaseVillager(target.getLocation().add(0, 1, 0), VillagerType.PLAINS);
+    }
+
+    public static void armorstand(Player p, Block target, String[] args) {
+        float pitch = Float.parseFloat(args[1]);
+        float yaw = Float.parseFloat(args[2]);
+        float roll = Float.parseFloat(args[3]);
+        MessageUtil.sendMessage(p, "&aPitch: %f, Yaw: %f, Roll: %f", pitch, yaw, roll);
+
+        ServerLevel level = ((CraftWorld) p.getWorld()).getHandle();
+        ArmorStand armorStand = new ArmorStand(level, target.getX(), target.getY() + 1, target.getZ());
+        armorStand.setNoGravity(true);
+        armorStand.setShowArms(true);
+        armorStand.setRightArmPose(new Rotations(pitch, yaw, roll));
+        armorStand.setItemSlot(EquipmentSlot.MAINHAND, CraftItemStack.asNMSCopy(new ItemStackBuilder(Material.IRON_AXE).build()));
+        level.addFreshEntity(armorStand, CreatureSpawnEvent.SpawnReason.CUSTOM);
+    }
+
+    public static void advancement(Player p, String[] args) {
+        Bukkit.getUnsafe().loadAdvancement(KeyUtils.newKey(""), "");
+        Advancement advancement = Bukkit.getAdvancement(KeyUtils.newKey(""));
```

Comment:
Oh this was an experimentation of mine that I forgot to delete (oops). But thanks for refactoring it into an utility class.

Reply:
Oh no problem 😅, I was wondering if that was the case but I was going to leave it in case it was just unfinished